main.py

```diff
@@ -30,7 +30,22 @@ def apply_custom_paths():
         for config_path in itertools.chain(*args.extra_model_paths_config):
             utils.extra_config.load_extra_path_config(config_path)
 
-    # --output-directory, --input-directory, --user-directory
+    # --temp-directory, --user-directory, --input-directory, --output-directory
+    if args.temp_directory:
+        temp_dir = os.path.join(os.path.abspath(args.temp_directory), "temp")
+        logging.info(f"Setting temp directory to: {temp_dir}")
+        folder_paths.set_temp_directory(temp_dir)
```

Comment:
Breaking change, but would be consistent with expectations of related args:

```suggestion
    if args.temp_directory:
        temp_dir = os.path.abspath(args.temp_directory)
        logging.info(f"Setting temp directory to: {temp_dir}")
        folder_paths.set_temp_directory(temp_dir)
```